docs/product-user-guide/src/main/asciidoc/drl-rules-other-con.adoc

```diff
@@ -0,0 +1,12 @@
+[id='drl-rules-other-con']
+= Other methods for creating DRL files
+
+As an alternative to creating and managing DRL files within the {CENTRAL} interface, you can create DRL files in external standalone projects using Red Hat Developer Studio, Java objects, or Maven archetypes. These standalone projects can then be integrated as knowledge JAR (kJAR) dependencies in existing {PRODUCT} projects in {CENTRAL}. The DRL files in your standalone project must contain at minimum the required `package` specification, `import` lists, and `rule` definitions. Any other DRL components, such as global variables and functions, are optional. All data objects related to a DRL rule must be included with your standalone DRL project or deployment.
```

Comment:
Okay @manstis, here's the updated preview for this section: http://file.rdu.redhat.com/~sterobin/BXMSDOC-2028/#drl-rules-other-con

The main change is the following to the intro paragraph:
"As an alternative to creating and managing DRL files within the Decision Central interface, you can create DRL files in external standalone projects using Red Hat Developer Studio, Java objects, or Maven archetypes. These standalone projects can then be integrated as knowledge JAR (kJAR) dependencies in existing Red Hat Decision Manager projects in Decision Central."

With the following at the end of each section (Dev Studio, Java, and Maven sections):
"To integrate the new rule assets with an existing project in Red Hat Decision Manager, you can compile the new project as a knowledge JAR (kJAR) and add it as a dependency in the pom.xml file of the project in Decision Central."

Two questions:
1) Does this work? I know there are details and other ways for integrating, but after speaking again with our CS, it was decided that we want include how to create the rules externally, and just mention briefly/roughly how to integrate, just so users at least know there are other options outside of Central. Then when we create a separate IDE doc or something that would address standalone projects, we'll provide all details and link to it from this doc.

2) Does this also apply to the Maven section (last of the "Other methods")? From what I understand, a kJAR is a maven project, or something. And I see that the Decision Central dependencies in the standalone Maven project are already documented in that section, for the Maven project's pom.xml file. I'm guessing it's an extra step for Maven projects, so I'm wondering if it's already a kJAR naturally, and if so, then maybe just need to add it as a dependency, nix the compile bit. I could be WAY off.

I'm just trying to get something accurate down, even if general.

Reply:
@sterobin New paragraph is fine with me.

As for your comment about "...the Maven section..." the steps in the example do result in a kJAR which could theoretically be executed by KIE Server/Decision Server "as is" without the need to add as a dependency to another project. However given you plan to create a separate IDE document I think it's best the details are left for then.

Reply:
Got it. Thanks.